sale_commission/model/invoice.py

```diff
@@ -0,0 +1,192 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (C) 2011 Pexego Sistemas Informáticos (<http://www.pexego.es>).
+#    All Rights Reserved
+#    $Id$
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+"""invoice agents"""
+
+from openerp import models, fields, api, _
+
+
+class invoice_line_agent(models.Model):
+    """invoice agents"""
+
+    _name = "invoice.line.agent"
+
+    invoice_line_id = fields.Many2one(
+        "account.invoice.line",
+        string="Invoice Line",
+        required=True,
+        ondelete="cascade"
+    )
+
+    invoice_id = fields.Many2one(
+        "account.invoice",
+        string="Invoice",
+        related="invoice_line_id.invoice_id"
+    )
+
+    invoice_date = fields.Date(
+        string="Invoice date",
+        related="invoice_id.date_invoice",
+        readonly=True
+    )
+
+    agent_id = fields.Many2one(
+        "sale.agent",
+        string="Agent",
+        required=True,
+        ondelete="cascade"
+    )
+
+    commission_id = fields.Many2one(
+        "commission",
+        string="Applied commission",
+        required=True,
+        ondelete="cascade",
+    )
+
+    settled = fields.Boolean(
+        string="Settled",
+        readonly=True,
+        default=False
+    )
+
+    quantity = fields.Float(string="Settled amount", default=0)
+
+    def calculate_commission(self, cr, uid, ids, context=None):
+        if context is None:
+            context = {}
+        for line_agent in self.browse(cr, uid, ids, context):
+            commission_type = line_agent.commission_id.type
+            if commission_type == 'fijo' and line_agent.commission_id.fix_qty:
+                subtotal = line_agent.invoice_line_id.price_subtotal
+                fix_qty = line_agent.commission_id.fix_qty
+                vals = {'quantity': subtotal * (fix_qty / 100.0)}
+                self.write(cr, uid, line_agent.id, vals, context=context)
+
+    @api.onchange("agent_id")
+    def do_set_commission_and_recalulate(self):
+        """Change commission and recalculate commissions
+        """
+        agent = self.agent_id
```

Comment:
agent and self.agent_id are a bit mixed. Could you use only `agent` after this line, it would show we are using only a single variable?